src/client.ts

```diff
@@ -191,12 +191,20 @@ import {
 } from './internal/utils/log';
 import { isEmptyObj } from './internal/utils/values';
 
+export interface AccessToken {
+  token: string;
+}
+export type TokenProvider = () => Promise<AccessToken>;
+
 export interface ClientOptions {
   /**
    * Defaults to process.env['OPENAI_API_KEY'].
    */
   apiKey?: string | undefined;
-
+  /**
+   * A function that returns a token to use for authentication.
+   */
+  tokenProvider?: TokenProvider | undefined;
```

Comment:
question: why call it `tokenProvider` and not `apiKeyProvider`? I think the former matches the existing terminology better